Model/Sync/Reset/Main.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace Yotpo\Core\Model\Sync\Reset;
+
+use Magento\Framework\App\Cache\TypeListInterface;
+use Magento\Framework\App\ResourceConnection;
+use Yotpo\Core\Model\Config;
+
+class Main
+{
+    const DELETE_LIMIT = 10000;
+    const UPDATE_LIMIT = 2000;
+
+    /**
+     * @var ResourceConnection
+     */
+    protected $resourceConnection;
+
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * @var TypeListInterface
+     */
+    protected $cacheTypeList;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        Config $config,
+        TypeListInterface $cacheTypeList
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->config = $config;
+        $this->cacheTypeList = $cacheTypeList;
+    }
+
+    /**
+     * @param int $storeId
+     * @param boolean $skipSyncTables
+     * @return void
+     * @throws \Zend_Db_Statement_Exception
+     */
+    public function resetSync($storeId, $skipSyncTables = false)
+    {
+        $this->setResetInProgressConfig($storeId, '1');
+        $this->deleteRunningCronSchedules();
+        if (!$skipSyncTables) {
+            $this->deleteAllFromTables($storeId);
+        }
```

Comment:
Why wouldn't we want to reset sync tables for all of the incoming reset executions?
I see we're clearing the sync table only for orders.

Reply:
It is the opposite case. We are not clearing the sync table for orders. We are clearing the table for all other entities. For orders we are passing $clearSyncTables=true at Yotpo\Core\Model\Sync\Reset::resetSync() method as we are calling parent::resetSync($storeId, true); there.

Why wouldn't we want to reset sync tables for all of the incoming reset executions?
Manaf : For the entities other than orders, we are managing the sync status by attribute flag. But for orders we are depending on a column "synced_to_yotpo_order" at sales_order table. So we are collecting the rows from this table where it's value=1 for a particular store_id and then deleting the matching rows at yotpo_orders_sync . This is needed because yotpo_orders_sync table doesn't have a store_id column in it. Hope you understood.